src/algebra/jordan/triple.lean

```diff
@@ -0,0 +1,215 @@
+(lebintz : ∀ (a b c d e: A), ⦃a, b, ⦃c, d, e⦄⦄  =
+  ⦃⦃a, b, c⦄, d, e⦄ - ⦃c, ⦃b, a, d⦄, e⦄ + ⦃c, d, ⦃a, b, e⦄⦄)
+
+/--
+We say that a pair of operators $(T,T^′)$ are Lebnitz if they satisfy a law reminiscent of
+differentiation.
+-/
+def lebnitz {A : Type*} [has_tp A] [has_add A] (T : A → A) (T'  : A → A) :=
+  ∀ (a b c : A),  T ⦃ a, b, c ⦄  = ⦃ T a, b, c⦄ + ⦃a, T' b, c⦄ + ⦃a, b, T c⦄
+
+namespace is_tp
+
+lemma radd {A : Type*} [has_tp A] [has_add A] [is_tp A] (a b c₁ c₂ : A) :
+  ⦃a, b, c₁ + c₂⦄ = ⦃a, b, c₁⦄ + ⦃a, b, c₂⦄ := by rw [comm, ladd, comm, comm c₂]
+
+variables {A : Type*} [has_tp A] [add_comm_group A] [is_tp A]
+
+lemma lzero (b c : A) : ⦃0, b, c⦄ = 0 :=
+(add_monoid_hom.mk' (λ (a : A), ⦃a, b, c⦄) (λ a₁ a₂, ladd a₁ a₂ b c )).map_zero
+
+lemma mzero (a c : A) : ⦃a, 0, c⦄ = 0 :=
+(add_monoid_hom.mk' (λ (b : A), ⦃a, b, c⦄) (λ b₁ b₂ , madd a b₁ b₂ c)).map_zero
+
+lemma rzero (a b : A) : ⦃a, b, 0⦄ = 0 :=
+by rw [comm, lzero]
+
+lemma lgsmul (a b c : A) (z : ℤ) : ⦃z•a, b, c⦄ = z•⦃a, b, c⦄ :=
+add_monoid_hom.map_zsmul ⟨λ (a : A), ⦃a, b, c⦄, lzero b c, λ _ _, ladd _ _ _ _⟩ _ _
+
+lemma mgsmul (a b c : A) (z : ℤ) : ⦃a, z•b, c⦄ = z•⦃a, b, c⦄ :=
+add_monoid_hom.map_zsmul ⟨λ (b : A), ⦃a, b, c⦄, mzero a c, λ _ _, madd _ _ _ _⟩ _ _
+
+lemma rgsmul (a b : A) (z : ℤ) (c : A) : ⦃a, b, z•c⦄ = z•⦃a, b, c⦄ :=
+  by rw [comm, lgsmul, comm]
+
+lemma lneg (a b c : A) : ⦃-a, b, c⦄ = -⦃a, b, c⦄ :=
+by rw [←sub_eq_zero, sub_neg_eq_add, ←ladd, neg_add_self, lzero]
+
+lemma mneg (a b c : A): ⦃a, -b, c⦄ = -⦃a, b, c⦄ :=
+by rw [←sub_eq_zero, sub_neg_eq_add, ←madd, neg_add_self, mzero]
+
+lemma rneg (a b c : A): ⦃a, b, -c⦄ = -⦃a, b, c⦄ := by rw [comm, lneg, comm]
+
+lemma lsub (a b c d : A) : ⦃a - d, b, c⦄ = ⦃a, b, c⦄ - ⦃d, b, c⦄ :=
+by rw [eq_sub_iff_add_eq, ← ladd, sub_add_cancel]
+
+lemma msub (a b c d : A) : ⦃a, b - c, d⦄ = ⦃a, b, d⦄ - ⦃a, c, d⦄ :=
+by rw [eq_sub_iff_add_eq, ← madd, sub_add_cancel]
+
+
+lemma rsub (a b c d : A) : ⦃a, b, c - d⦄ = ⦃a, b, c⦄ - ⦃a, b, d⦄ :=
+by rw [comm, lsub, comm, comm d]
+
+lemma lr_bilinear (a₁ a₂ b c₁ c₂ : A) : ⦃a₁ + a₂, b, c₁ + c₂⦄ =
+  ⦃a₁, b, c₁⦄ + ⦃a₁, b, c₂⦄ + ⦃a₂, b, c₁⦄ + ⦃a₂, b, c₂⦄ :=
+calc ⦃a₁ + a₂, b, c₁ + c₂⦄ = ⦃a₁, b, c₁ + c₂⦄ + ⦃a₂, b, c₁ + c₂⦄ : by rw ladd
+... = ⦃a₁, b, c₁⦄ + ⦃a₁, b, c₂⦄ + ⦃a₂, b, c₁ + c₂⦄ : by rw radd
+... = ⦃a₁, b, c₁⦄ + ⦃a₁, b, c₂⦄ + (⦃a₂, b, c₁⦄ + ⦃a₂, b, c₂⦄) : by rw radd
+... = ⦃a₁, b, c₁⦄ + ⦃a₁, b, c₂⦄ + ⦃a₂, b, c₁⦄ + ⦃a₂, b, c₂⦄ : by rw ← add_assoc
+
+lemma polar (a b c : A) : ⦃a + c, b, a + c⦄ = ⦃a, b, a⦄ + 2•⦃a, b, c⦄ + ⦃c, b, c⦄ :=
+calc ⦃a + c, b, a + c⦄ = ⦃a, b, a⦄ + ⦃a, b, c⦄ + ⦃c, b, a⦄ + ⦃c, b, c⦄ :
+  by rw lr_bilinear a c b a c
+... = ⦃a, b, a⦄ + ⦃a, b, c⦄ + ⦃a, b, c⦄ + ⦃c, b, c⦄ : by rw comm c b a
+... = ⦃a, b, a⦄ + (⦃a, b, c⦄ + ⦃a, b, c⦄)  + ⦃c, b, c⦄ : by rw ← add_assoc
+... = ⦃a, b, a⦄ + 2•⦃a, b, c⦄ + ⦃c, b, c⦄ : by rw two_nsmul
+
+/-- Define the multiplication operator `D` -/
+@[simps] def D (a b : A) : add_monoid.End A :=
```

Comment:
This is a pretty bad name! Isn't there anything more explicit in the literature?

Reply:
It's usually written $D(a,b)$ in LaTeX (occasionally $L(a,b)$ or $a \Box b$). Normally it isn't introduced with a name, I got the term "multiplication operator" from N.A. Altwaijry and C.M. Edwards, The Banach-Lie algebra of multiplication operators on a JBW∗ -triple. J. Funct. Anal. 254 (2008), 2866-2892.

Reply:
`mul` or `mult` would already be a better name. You can't just use a letter.

Reply:
Fully it's `is_tp.D`. If a symbol like `+`, `*` etc can represent a bilinear map, why not `D`?

Reply:
`+` and `*` are notation; feel free to introduce `D` as local notation or an abbreviation. But the convention is that the name is descriptive.

Reply:
@kbuzzard, you say this, but we have `algebraic_geometry.AffineScheme.Γ`! I think that one is due a rename to `.Gamma` or something.

Reply:
We also have `polynomial.C`, `polynomial.X`.